Neos.Neos/Classes/FrontendRouting/Options.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\FrontendRouting;
+
+/**
+ * Immutable filter DTO for {@see NodeUriBuilder::uriFor()}
+ *
+ * Example:
+ *
+ *   Options::create(forceAbsolute: true);
+ *
+ * @api for the factory methods; NOT for the inner state.
+ */
+final readonly class Options
+{
+    /**
+     * @internal the properties themselves are readonly; only the write-methods are API.
+     * @param array<string, mixed> $routingArguments
+     */
+    private function __construct(
+        public bool $forceAbsolute,
+        public string $format,
+        public array $routingArguments,
+    ) {
+    }
+
+    /**
+     * Creates an instance with the specified options
+     *
+     * Note: The signature of this method might be extended in the future, so it should always be used with named arguments
+     * @see https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments
+     */
+    public static function create(
+        bool $forceAbsolute = null
+    ): self {
+        return new self(
+            $forceAbsolute ?? false,
+            '',
+            []
+        );
+    }
+
+    /**
+     * Returns a new instance with the specified additional options
+     *
+     * Note: The signature of this method might be extended in the future, so it should always be used with named arguments
+     * @see https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments
+     */
+    public function with(
+        bool $forceAbsolute = null
+    ): self {
```

Comment:
these signatures dont feel quite right because they only have one argument ... and i wonder if we ever want to add any. Maybe rather

```
Options::forceAbsolute()
```

?

Reply:
I agree. Or rather sth like `withEnforcedAbsoluteUri()` to be consistent

Reply:
Went for 

```
Options::createForceAbsolute()
```

and 

```
Options::createEmpty()->withForceAbsolute();
```

felt the most aligned with `withCustomFormat` and `withCustomRoutingArguments`